trading/exchanges/websockets_exchanges/websocket/cryptofeed_callbacks.py

```diff
@@ -0,0 +1,95 @@
+#  This library is free software; you can redistribute it and/or
+#  modify it under the terms of the GNU Lesser General Public
+#  License as published by the Free Software Foundation; either
+#  version 3.0 of the License, or (at your option) any later version.
+#
+#  This library is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#  Lesser General Public License for more details.
+#
+#  You should have received a copy of the GNU Lesser General Public
+#  License along with this library.
+import time
+
+from tools import get_logger
+
+
+class CryptoFeedCallBack:
+    def __init__(self, parent):
+        self.parent = parent
+
+        self.logger = get_logger(f"WebSocket"
+                                 f" - {self.parent.exchange_manager.exchange.get_name()}"
+                                 f" - {self.__class__.__name__}")
+
+
+class OrderBookCallBack(CryptoFeedCallBack):
+    def __init__(self, parent):
+        super().__init__(parent)
+
+    async def l2_order_book_callback(self, feed, pair, book, timestamp):
+        symbol_data = self.parent.get_symbol_data_from_pair(pair)
+        if symbol_data and symbol_data.order_book_is_initialized():
+            symbol_data.update_order_book(self.parent.convert_into_ccxt_full_order_book(symbol_data.symbol,
+                                                                                        book,
+                                                                                        timestamp))
+
+
+class RecentTradesCallBack(CryptoFeedCallBack):
+    def __init__(self, parent):
+        super().__init__(parent)
+
+    async def recent_trades_callback(self, feed, pair, order_id, timestamp, side, amount, price):
+        symbol_data = self.parent.get_symbol_data_from_pair(pair)
+        if symbol_data and symbol_data.recent_trades_are_initialized():
+            symbol_data.add_new_recent_trades(self.parent.convert_into_ccxt_recent_trade(symbol_data.symbol,
+                                                                                         side,
+                                                                                         amount,
+                                                                                         price,
+                                                                                         timestamp))
+
+
+class TickersCallBack(CryptoFeedCallBack):
+    def __init__(self, parent):
+        super().__init__(parent)
+
+    async def tickers_callback(self, feed, pair, bid, ask, timestamp):
+        # self.logger.info(f'TICKER / Pair: {pair} Bid: {bid} Ask: {ask}')
+        # symbol_data = self.parent.get_symbol_data_from_pair(pair)
+        # if symbol_data:
+        #     symbol_data.update_symbol_ticker(
+        #         self.parent.convert_into_ccxt_ticker(symbol_data.symbol, bid, ask, timestamp))
+        pass
```

Comment:
to be implemented later ?

Reply:
I am working on it, it's the reason of the "WIP" ;)